toolchest_client/tools/tool.py

```diff
@@ -264,24 +274,26 @@ def _generate_jobs(self, should_run_in_parallel):
                 for _, file_path in adjusted_input_file_paths:
                     # This is assuming only one input file per parallel run.
                     # This will need to be changed once we support multiple input files for parallelization.
-                    yield [file_path]
+                    yield [file_path, PARALLEL_FILE_IS_IN_S3]
             else:
                 # Grouped parallelization. Right now, this only supports grouping by R1/R2 for paired-end inputs
                 input_file_paths_pairs = split_paired_files_by_lines(
                     input_file_paths=self.input_files,
                     max_bytes=self.max_input_bytes_per_node,
                 )
                 for input_file_path_pair in input_file_paths_pairs:
-                    yield input_file_path_pair
+                    yield input_file_path_pair, PARALLEL_FILE_IS_IN_S3
 
         else:
             # Make sure we're below plan/multi-part limit for non-splittable files
-            for file_path in self.input_files:
-                check_file_size(file_path, max_size_bytes=FOUR_POINT_FIVE_GIGABYTES)
+            for file_path, file_is_in_s3 in zip(self.input_files, self.inputs_are_in_s3):
+                if not file_is_in_s3:
+                    check_file_size(file_path, max_size_bytes=FOUR_POINT_FIVE_GIGABYTES)
+                # If the file is already in S3, no need to check max size.
```

Comment:
Should we guard against large input files from S3? Could cause us to exceed resource limits (e.g. a 200 GB input file causing an out of memory issue for Kraken 2)

Reply:
Good point -- we can merge this in for now and implement later.

Reply:
To clarify: is there any guard for the 5 GB limit on S3 files right now?

Reply:
Not right now. I'll update this with a quick fix PR later today.